src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/locations/availability/DateTimeAvailabilityService.kt

```diff
@@ -80,3 +93,31 @@ class DateTimeAvailabilityService(
     return AvailabilityStatus.SHARED
   }
 }
+
+class DateTimeLocationsBuilder private constructor() : LocationAvailabilityService.AvailableLocationBuilder() {
```

Comment:
Each service has it's own builder.  This keeps the logic cleaner with regards to the build list of available locations.  E.g. the sorting for date time is different to that of time slot.